crates/stages/stages/src/stages/merkle.rs

```diff
@@ -344,18 +344,18 @@ where
 
 /// Check that the computed state root matches the root in the expected header.
 #[inline]
-fn validate_state_root(
+fn validate_state_root<H: BlockHeader + Debug>(
```

Comment:
```suggestion
fn validate_state_root<H: BlockHeader>(
```

Reply:
we need / want the `Debug` here for the error log on invalid state root

Reply:
this seems like something we should also upstream to alloy

Reply:
ah I didn't realize this is alloy's BlockHeader